File: DesignPatternsExercises/java/VisitorDesignPattern/Solution/Customer.java
import java.util.ArrayList;

public class Customer implements Visitable {
    private String name;
    // private ArrayList<__________(1)__________> orders = new ArrayList<>();
    private ArrayList<Order> orders = new ArrayList<>();

    public Customer(String name) {
        this.name = name;
    }

    @Override
    public void accept(Visitor visitor) {
        // visitor.__________(2)__________;
        visitor.visit(this);
        System.out.println("Orders:");
        for (Order order : orders)
            // order.__________(3)__________;
            order.accept(visitor);
        System.out.println();
    }

    public void addOrder(Order order) {
        orders.add(order);
    }

    public String getName() {
        return name;
    }
}

File: DesignPatternsExercises/ruby/visitor_design_pattern/solution/customer.rb
require_relative 'visitable'

class Customer < Visitable
  attr_reader :name

  # def __________(1)__________(name)
  def initialize(name)
    @name = name
    # __________(2)__________ = []
    @orders = []
  end

  # def __________(3)__________(visitor)
  def accept(visitor)
    # visitor.__________(4)__________
    visitor.visit(self)
    puts 'Orders:'
    # @orders.each { |order| order.__________(5)__________ }
    @orders.each { |order| order.accept(visitor) }
    puts
  end

  def add_order(order)
    @orders << order
  end
end
